src/components/NavBar/Buttons.js

```diff
@@ -0,0 +1,210 @@
+      isLast
+        ? spacing.gutters.mediumAndUp
+        : parseInt(spacing.normal, 10) / 2}px;
+    padding-left: ${({ isFirst }) =>
+      isFirst
+        ? spacing.gutters.mediumAndUp
+        : parseInt(spacing.normal, 10) / 2}px;
+  `};
+`;
+
+const Anchor = BaseButton.withComponent("a");
+
+const getButtonColor = isInverted =>
+  isInverted ? colors.onyx.base : colors.white.base;
+
+const Button = ({ children, href, target, rel, ...props }) => {
+  if (href) {
+    return (
+      <Anchor
+        {...props}
+        href={href}
+        target={target}
+        rel={getRelByTarget(target, rel)}
+      >
+        <Consumer>
+          {isInverted =>
+            children && children({ color: getButtonColor(isInverted) })
+          }
+        </Consumer>
+      </Anchor>
+    );
+  }
+
+  return (
+    <BaseButton type="button" {...props}>
+      <Consumer>
+        {isInverted =>
+          children &&
+          children({ color: getButtonColor(isInverted), isInverted })
+        }
+      </Consumer>
+    </BaseButton>
+  );
+};
+
+Button.propTypes = {
+  children: PropTypes.func,
+  href: PropTypes.string,
+  target: PropTypes.string,
+  rel: PropTypes.string,
+  isFirst: PropTypes.bool,
+  isLast: PropTypes.bool
+};
+
+Button.defaultProps = {
+  children: null,
+  href: null,
+  isFirst: false,
+  isLast: false,
+  target: "_self",
+  rel: null
+};
+
+const Heading = styled.h1`
+  padding: 0;
+  margin: 0;
+  font-size: 1em;
+`;
+
+const LogoBtn = styled(Button)`
+  ${largeAndUp`
+    padding-left: ${spacing.gutters.mediumAndUp}px !important;
+  `};
+`;
+
+const LogoContainer = ({ children, href, isHeading, ...props }) => (
+  <LogoBtn href={href} {...props}>
+    {isHeading
+      ? (...args) => <Heading>{children && children(...args)}</Heading>
+      : children}
+  </LogoBtn>
+);
+
+LogoContainer.propTypes = {
+  children: PropTypes.func,
+  href: PropTypes.string,
+  isHeading: PropTypes.bool
+};
+
+LogoContainer.defaultProps = {
+  children: () => null,
+  href: null,
+  isHeading: false
+};
+
+const MenuButton = ({ children, ...props }) => (
+  <Button isFirst {...props}>
+    {({ color }) => (
+      <React.Fragment>
+        <Menu size={20} color={color} />
+        {children}
+      </React.Fragment>
+    )}
+  </Button>
+);
+
+MenuButton.propTypes = {
+  children: PropTypes.node
+};
+
+MenuButton.defaultProps = {
+  children: null
+};
+
+const SearchButton = ({ children, ...props }) => (
+  <Button {...props}>
+    {({ color }) => (
+      <React.Fragment>
+        <Search size={21} color={color} />
+        {children}
+      </React.Fragment>
+    )}
+  </Button>
+);
+
+SearchButton.propTypes = {
+  children: PropTypes.node
+};
+
+SearchButton.defaultProps = {
+  children: null
+};
+
+const UserButton = ({ children, ...props }) => (
+  <Button {...props}>
+    {({ color }) => (
+      <React.Fragment>
+        <User size={24} color={color} />
+        {children}
+      </React.Fragment>
+    )}
+  </Button>
+);
```

Comment:
These buttons with icons seem a bit repetitive. Maybe we should create an `IconButton` component?